File: test/notification_icon_data_test.dart
import 'dart:ui';

import 'package:flutter_test/flutter_test.dart';
import 'package:geofence_foreground_service/constants/json_keys.dart';
import 'package:geofence_foreground_service/models/notification_icon_data.dart';

void main() {
  test('toJson maps enums and optional color', () {
    const icon = NotificationIconData(
      resType: ResourceType.mipmap,
      resPrefix: ResourcePrefix.ic,
      name: 'launcher',
    );
    final json = icon.toJson();
    expect(json[JsonKeys.resType], 'mipmap');
    expect(json[JsonKeys.resPrefix], 'ic');
    expect(json[JsonKeys.name], 'launcher');
    expect(json[JsonKeys.backgroundColorRgb], isNull);
  });

  test('toJson includes RGB when backgroundColor set', () {
    final icon = NotificationIconData(
      resType: ResourceType.drawable,
      resPrefix: ResourcePrefix.img,
      name: 'bell',
      backgroundColor: const Color(0xAABBCCDD),
    );
    final json = icon.toJson();
    expect(json[JsonKeys.backgroundColorRgb], isNotNull);
    expect(json[JsonKeys.backgroundColorRgb] as String, contains(','));
  });
}

File: android/src/test/kotlin/com/f2fk/geofence_foreground_service/models/NotificationIconDataTest.kt
package com.f2fk.geofence_foreground_service.models

import com.f2fk.geofence_foreground_service.Constants
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test

class NotificationIconDataTest {

    @Test
    fun fromJson_mapsFields() {
        val json = mapOf<String, Any>(
            Constants.resType to "mipmap",
            Constants.resPrefix to "ic",
            Constants.name to "launcher",
            Constants.backgroundColorRgb to "1,2,3",
        )
        val data = NotificationIconData.fromJson(json)
        assertEquals("mipmap", data.resType)
        assertEquals("ic", data.resPrefix)
        assertEquals("launcher", data.name)
        assertEquals("1,2,3", data.backgroundColorRgb)
    }

    @Test
    fun fromJson_nullBackgroundColor() {
        val json = mapOf<String, Any>(
            Constants.resType to "drawable",
            Constants.resPrefix to "img",
            Constants.name to "bell",
        )
        val data = NotificationIconData.fromJson(json)
        assertNull(data.backgroundColorRgb)
    }
}
